discovery_engine_core/providers/src/client.rs

```diff
@@ -119,15 +119,24 @@ where
 
 impl<T> Seal for NewsQuery<'_, T> {}
 
-/// Parameters determining which headlines to fetch
+/// Parameters determining which headlines to fetch.
 pub struct HeadlinesQuery<'a> {
-    /// Common parts
+    /// Common parts.
     pub common: CommonQueryParts<'a>,
+    /// Favourite sources.
+    pub sources: &'a [String],
 }
 
 impl Query for HeadlinesQuery<'_> {
     fn setup_url(&self, url: &mut Url) -> Result<(), Error> {
-        self.common.setup_url(url, "_lh")
+        self.common.setup_url(url, "_lh")?;
+
+        if !self.sources.is_empty() {
+            let mut query = url.query_pairs_mut();
+            query.append_pair("sources", &self.sources.join(","));
+        };
```

Comment:
:orange_circle:  As far as I can tell this should go into `CommonQueryParts` as it applies to both news and headlines.

Through the issue only says `HeadlineQuery` :thinking: 

@acrrd is meant to be applied to both kinds of queries?

Reply:
Now we plan to use it only for query to the headline endpoints but we can indeed add it to the common query so it is already there we decide to do something different later.